aries_vcx/src/protocols/connection/invitee/state_machine.rs

```diff
@@ -242,17 +242,9 @@ impl SmConnectionInvitee {
         let Self { state, .. } = self;
         let thread_id = invitation.get_id()?;
         let state = match state {
-            InviteeFullState::Initial(state) => InviteeFullState::Invited(
-                (
-                    state.clone(),
-                    invitation,
-                    state.did_doc.ok_or(AriesVcxError::from_msg(
-                        AriesVcxErrorKind::InvalidState,
-                        "Expected none None state.did_doc result given current state",
-                    ))?,
-                )
-                    .into(),
-            ),
+            InviteeFullState::Initial(state) => {
```

Comment:
Can you please extract the changes related to cleaning up the initial/invited state and related APIs a small separate PR? 
For sake of better changelog, plus it's something that can be merged probably very soon

Reply:
In fact, I'm separating the code completely so that I don't interfere with current code. I'll revert the changes on these files, as initially I was trying to use the same state structs.